File: lib/modules/family/controllers/family_controller.dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'package:get_storage/get_storage.dart';
import '../../../core/network/api_client.dart';

class FamilyController extends GetxController {
  final familyNameController = TextEditingController();
  final joinFamilyIdController = TextEditingController();

  var isLoading = false.obs;
  var currentFamilyId = ''.obs;

  @override
  void onInit() {
    super.onInit();
    // Load existing family if available in local storage
    currentFamilyId.value = GetStorage().read('family_id') ?? '';
  }

  Future<void> createFamily() async {
    final name = familyNameController.text.trim();
    if (name.isEmpty) return;

    isLoading.value = true;
    try {
      final userId = GetStorage().read('user_id');
      final data = await ApiClient().post('/family/create', {
        'userId': userId,
        'name': name,
        'avatar':
            'https://via.placeholder.com/150', // Replace with real asset later
      });

      if (data['success'] == true) {
        Get.snackbar('Success', 'Family created successfully!',
            backgroundColor: Colors.green, colorText: Colors.white);
        currentFamilyId.value = data['data']['familyId'];
        GetStorage().write('family_id', currentFamilyId.value);
        familyNameController.clear();
      } else {
        Get.snackbar('Error', data['message'] ?? 'Failed to create family',
            backgroundColor: Colors.redAccent, colorText: Colors.white);
      }
    } finally {
      isLoading.value = false;
    }
  }

  Future<void> joinFamily() async {
    final familyId = joinFamilyIdController.text.trim();
    if (familyId.isEmpty) return;

    isLoading.value = true;
    try {
      final userId = GetStorage().read('user_id');
      final data = await ApiClient().post('/family/join', {
        'userId': userId,
        'familyId': familyId,
      });

      if (data['success'] == true) {
        Get.snackbar('Success', 'Joined family successfully!',
            backgroundColor: Colors.green, colorText: Colors.white);
        currentFamilyId.value = familyId;
        GetStorage().write('family_id', currentFamilyId.value);
        joinFamilyIdController.clear();
      } else {
        Get.snackbar('Error', data['message'] ?? 'Failed to join family',
            backgroundColor: Colors.redAccent, colorText: Colors.white);
      }
    } finally {
      isLoading.value = false;
    }
  }

  Future<void> leaveFamily() async {
    isLoading.value = true;
    try {
      final userId = GetStorage().read('user_id');
      final data = await ApiClient().post('/family/leave', {
        'userId': userId,
      });

      if (data['success'] == true) {
        Get.snackbar('Success', 'You have left the family.',
            backgroundColor: Colors.green, colorText: Colors.white);
        currentFamilyId.value = '';
        GetStorage().remove('family_id');
      } else {
        Get.snackbar('Error', data['message'] ?? 'Failed to leave family',
            backgroundColor: Colors.redAccent, colorText: Colors.white);
      }
    } finally {
      isLoading.value = false;
    }
  }
}

File: arvind-party-backend/src/controllers/familyController.js
const User = require('../models/User');
const Family = require('../models/Family');

exports.getMyFamily = async (req, res) => {
  try {
    const userId = req.user.id || req.user.userId;
    const family = await Family.findOne({ members: userId }).populate('members', 'name avatar').populate('leader', 'name avatar');
    
    if (family) {
      res.status(200).json({ success: true, family, message: "Family data loaded" });
    } else {
      res.status(200).json({ success: true, family: null, message: "No family joined yet" });
    }
  } catch (error) {
    console.error('Get Family Error:', error);
    res.status(500).json({ success: false, message: 'Failed to load family data' });
  }
};

exports.createFamily = async (req, res) => {
  try {
    const { name, logo } = req.body;
    const userId = req.user.id || req.user.userId;
    
    const newFamily = await Family.create({ name, logo, leader: userId, members: [userId] });
    res.status(201).json({ success: true, family: newFamily, message: `Family '${name}' created successfully!` });
  } catch (error) {
    console.error('Create Family Error:', error);
    res.status(500).json({ success: false, message: 'Failed to create family' });
  }
};

exports.joinFamily = async (req, res) => {
  try {
    const { familyId } = req.body;
    const userId = req.user.id || req.user.userId;

    const family = await Family.findByIdAndUpdate(familyId, { $addToSet: { members: userId } }, { new: true });
    if (!family) {
      return res.status(404).json({ success: false, message: 'Family not found' });
    }
    res.status(200).json({ success: true, family, message: 'Joined family successfully' });
  } catch (error) {
    console.error('Join Family Error:', error);
    res.status(500).json({ success: false, message: 'Failed to join family' });
  }
};